x-pack/plugins/security_solution/public/resolver/view/use_sync_selected_node.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext, useCallback, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import * as selectors from '../store/selectors';
+import * as nodeModel from '../../../common/endpoint/models/node';
+import { SideEffectContext } from './side_effect_context';
+import { ResolverState } from '../types';
+import { ResolverAction } from '../store/actions';
+
+/**
+ * This custom hook, will maintain the state of the active/selected node with the what the selected nodeID is in url state.
+ * This means page refreshes, direct links, back and forward buttons, should always pan to the node defined in the url
+ * In the scenario where the nodeList is visible in the panel, there is no selectedNode, but this would naturally default to the origin node based on `serverReturnedResolverData` on initial load and refresh
+ * This custom hook should only be called once on resolver load, following that the url nodeID should always equal the selectedNode.  This is currently called in `resolver_without_providers.tsx`.
+ */
+export function useSyncSelectedNode() {
+  const dispatch: (action: ResolverAction) => void = useDispatch();
+  const { timestamp } = useContext(SideEffectContext);
+  const currentPanelParameters = useSelector(selectors.panelViewAndParameters);
+  const selectedNode = useSelector(selectors.selectedNode);
+
+  // Get all the nodeIDs to confirm that the node we are attempting to pan to actually exists
+  const allNodeIDs: string[] = useSelector(
+    useCallback((state: ResolverState) => {
```

Comment:
Instead of grabbing all the `nodeIDs` would it make sense to use this selector?
https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/resolver/store/data/selectors.ts#L442-L454

It handles adding them to a map already I believe. That way we don't have to do `allNodeIDs.includes` either. We can just use that to do the map lookup.

Reply:
Thanks! I was being lazy about finding a proper map for this. Anyways, I'll update and then add tests for all of it

Reply:
Fixed this, but closing and opening against Kibana once I've added tests